### PR TITLE
ncl Chapter 13

### PR DESCRIPTION
**Pull request by NormanLittle**

### CODE REVIEW

**Comment by utopia-2** [comment]
:+1: for continuing Java Training :smile:.

File: src/com/selftest/stack/StackImpl.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+public class StackImpl<T> implements Stack<T> {
+
+	private T[] stack;
+	private int size;
+
+	@SafeVarargs
+	public StackImpl( int size, T ... values ) {
+		stack = Arrays.copyOf( values, size );
+		size = 0;
```
Comment (utopia-2) [inline src/com/selftest/stack/StackImpl.java:17]: If the user were to add values to the constructor, would they expect this to discard them?

Comment (NormanLittle) [inline src/com/selftest/stack/StackImpl.java:17]: Hmm, I would imagine that would be a surprise feature.  I forgot to update the index (size)...but I am sure I _would_ have picked that up in unit testing :)

**Comment by NormanLittle** [comment]
@Utopia1138 I couldn't possibly stop Java training _before_ we get to Lambdas!  Sadly, the new job is only on JDK 1.7 so I'll have to wait a wee while to use 'em in actual work.

File: src/com/selftest/StackDemo.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	public static void main( String[] args ) {
+		Stack<Character> stack = new StackImpl<Character>( 8 );
+		try {
+			for ( char c = 'a' ; c <= 'z' ; c++ )
+				stack.push( c );
+		}
+		catch ( StackFullException e ) {
+			System.out.printf( "%s: [%s]%n",
+			    							 e.getMessage(),
+			    							 stack.stream().map( c -> c.toString() ).collect( Collectors.joining( "," ) ) );
```
Comment (Spinodal) [inline src/com/selftest/StackDemo.java:22]: Wow the formatting went a bit mental here!

File: answers.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+7)  Yes.
+```java
+static void doSomething( SomeValue<? extends Number> ) {
+}
+```
+
+8)
+```java
+public static <T> T MyGen( T t) {
+  return t;
```
Comment (NormanLittle) [inline answers.md:36]: yay! Thanks for the 'gentleman octopus'.  Disappointed to miss the meeting this morning.

**Pull request closed without merging**